test/server/cards/01_SOR/units/HanSoloReluctantHero.spec.ts

```diff
@@ -0,0 +1,80 @@
+describe('Han Solo Reluctant Hero', function() {
+    integration(function(contextRef) {
+        describe('Han Solo Reluctant Hero\'s ability', function() {
+            beforeEach(function () {
+                contextRef.setupTest({
+                    phase: 'action',
+                    player1: {
+                        hand: ['han-solo#reluctant-hero'],
+                    },
+                    player2: {
+                        groundArena: [{ card: 'wampa', upgrades: ['shield'] }, 'consular-security-force']
+
+                    }
+                });
+            });
+
+            it('while attacking an enemy ground unit, should deal damage to the defender before taking damage.', function () {
+                const { context } = contextRef;
+                const reset = () => {
+                    context.hanSolo.exhausted = false;
+                    context.hanSolo.damage = 0;
+                    context.consularSecurityForce.damage = 0;
+                    context.consularSecurityForce.exhausted = false;
+                };
+                // Case 1 attack action shouldn't deal damage to the shielded wampa and should deal 4 damage to Han Solo
+                context.player1.clickCard(context.hanSolo);
+                context.player1.clickPrompt('Ambush');
+                context.player1.clickCard(context.wampa);
+
+                // check board state
+                expect(context.hanSolo.damage).toBe(4);
+                expect(context.wampa.damage).toBe(0);
+
+                // reset board
+                reset();
+
+                context.player2.passAction();
+                // Case 2 attack action should defeat wampa and no damage on hansolo
+                context.player1.clickCard(context.hanSolo);
+                context.player1.clickCard(context.wampa);
+
+                // check board state
+                expect(context.hanSolo.damage).toBe(0);
+                expect(context.wampa.location).toBe('discard');
+
+                // Case 3 attack action should deal 6 damage to consular security force and 3 on han solo
+                context.player2.passAction();
+                context.hanSolo.exhausted = false;
+                context.player1.clickCard(context.hanSolo);
+                context.player1.clickCard(context.consularSecurityForce);
+
+                // check board state
+                expect(context.consularSecurityForce.damage).toBe(6);
+                expect(context.hanSolo.damage).toBe(3);
+
+                // reset board
+                reset();
+
+                // Case 4 check if Han Solo takes damage
```

Comment:
I don't this case tests what we need, since Han does not defeat this unit he would take damage even if the test failed. Need to test with a unit that he defeats to make sure that he's not dealing damage first when defending